Objects/stringobject.c

```diff
@@ -1209,8 +1209,11 @@ string_richcompare(PyStringObject *a, PyStringObject *b, int op)
     PyObject *result;
 
     /* Make sure both arguments are strings. */
-    if (!(PyString_Check(a) && PyString_Check(b))) {
+     if (!(PyString_Check(a) && PyString_Check(b)) ) {
         result = Py_NotImplemented;
+        if (PyErr_WarnPy3k("comparing unicode and byte strings has different semantics in 3.x: convert the second string to byte.", 1) < 0) {
+                goto out;
```

Comment:
`goto out` indented 1 level too many?

Reply:
I have added the condition to the existing if.